cpp/include/cudf/copying.hpp

```diff
@@ -444,6 +444,25 @@ std::vector<contiguous_split_result> contiguous_split(cudf::table_view const& in
                                                       std::vector<size_type> const& splits,
                                                       rmm::mr::device_memory_resource* mr = rmm::mr::get_default_resource());
 
+struct packed_table {
+    struct serialized_column {
+        cudf::data_type _type;
+        cudf::size_type _size;
+        size_t _data_offset;
+        size_t _null_mask_offset;
+        cudf::size_type _num_children;
+    };
+
+    std::vector<serialized_column> table_metadata;
+    std::unique_ptr<rmm::device_buffer> table_data;
+};
+
+packed_table pack(cudf::table_view const& input,
+                  rmm::mr::device_memory_resource* mr = rmm::mr::get_default_resource());
+
+contiguous_split_result unpack(packed_table & input,
```

Comment:
Not true. The table data buffer from `packed_table` is `std::move`d out into the returning `contiguous_split_result`. This is done to reduce memory usage. I can change it to pointer.

Reply:
This should be a `unique_ptr<packed_table>` then to indicate that the function is taking ownership of the `packed_table`. Furthermore, there will need to be a constructor to allow the user to construct a `packed_table` from the host/device buffer.

Reply:
A `packed_table` should only be allowed to be generated by `cudf::pack`. Unless you mean there should be a method to convert `packed_table` returned by `cudf::pack` into a `unique_ptr<packed_table>`. Then we're just transferring the take-by-ref API design from `unpack` to that method.

There's two more ways this can be handled:
1. Return `unique_ptr<packed_table>` from `pack`.
2. Take by `packed_table ` by rvalue reference in `unpack`.

Reply:
> A packed_table should only be allowed to be generated by cudf::pack

I think you're forgetting about when sending a `packed_table` over the wire. On the receive side, all there is is a host buffer/device_buffer. The receiver needs to be able to construct a `packed_table` from those buffers, and then pass ownership of those buffers to `cudf::unpack`.